### PR TITLE
Fix/missing projects in list 2

### PR DESCRIPTION
**Pull request by BCerki**
Card: https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/bcgov/cas-cif/1106

### CODE REVIEW

File: schema/deploy/computed_columns/project_manager_names_concatenated.sql
Comment (dleard) [inline schema/deploy/computed_columns/project_manager_names_concatenated.sql:16]: I think using `string_agg` instead of concat here will simplify things.
https://www.postgresqltutorial.com/postgresql-aggregate-functions/postgresql-string_agg-function/

File: schema/deploy/computed_columns/project_manager_names_concatenated.sql
Comment (dleard) [inline schema/deploy/computed_columns/project_manager_names_concatenated.sql:14]: hardcoding this index here handcuffs us a bit if we add/remove manager labels in the future. See below comment on how we can filter the results more dynamically.

File: schema/deploy/computed_columns/project_manager_names_concatenated.sql
Comment (dleard) [inline schema/deploy/computed_columns/project_manager_names_concatenated.sql:21]: We only want primary managers here. I'd suggest filtering on the id's where the label name contains `primary`
```suggestion
                and cif.project_manager.project_manager_label_id in (select id from cif.project_manager_label where label ilike '%primary%')))))::text);
```

File: schema/deploy/computed_columns/project_manager_names_concatenated.sql
Comment (dleard) [inline schema/deploy/computed_columns/project_manager_names_concatenated.sql:25]: now that we don't need variables or loops, we can change this function to a pure `sql` function (and add `stable` so it recognizes it as a computed column
```suggestion
$$ language sql stable;
```

File: schema/deploy/computed_columns/project_manager_names_concatenated.sql
Comment (dleard) [inline schema/deploy/computed_columns/project_manager_names_concatenated.sql:27]:
```suggestion
grant execute on function cif.project_primary_managers to cif_internal, cif_admin;
```

File: schema/deploy/computed_columns/project_manager_names_concatenated.sql
Comment (dleard) [inline schema/deploy/computed_columns/project_manager_names_concatenated.sql:7]: I think `cif.project_primary_managers` is a more explicit name for this function
```suggestion
create or replace function cif.project_primary_managers(cif.project)
```

File: schema/deploy/computed_columns/project_manager_names_concatenated.sql
Comment (dleard) [inline schema/deploy/computed_columns/project_manager_names_concatenated.sql:29]:
```suggestion
comment on function cif.project_primary_managers is 'Returns a concatenated list of the project managers (used for filtering the projects table)';
```

**Review by dleard — CHANGES_REQUESTED** [review]

**Review by sam-warren — DISMISSED** [review]
Looks good - might want to wait for @Sepehr-Sobhani to review before go-ahead as I haven't spent as much time on the project though 😄

**Comment by Sepehr-Sobhani** [comment]
Looks great to me, Thank you @BCerki!
I just noticed two issues:

1. when I remove a user from being a primary manager, I still get that project when I filter projects based on that user:

https://www.loom.com/share/19f8611a8a904e0cb7eee95cc28c0943

2. When I add a project  I can't see the project in the project list if it doesn't have a primary manager (it's either without a manager or with a secondary manager) -> not sure if this is related to your changes.

**Comment by BCerki** [comment]
> Looks great to me, Thank you @BCerki! I just noticed two issues:
> 
> 1. when I remove a user from being a primary manager, I still get that project when I filter projects based on that user:
> 
> https://www.loom.com/share/19f8611a8a904e0cb7eee95cc28c0943
> 
> 2. When I add a project  I can't see the project in the project list if it doesn't have a primary manager (it's either without a manager or with a secondary manager) -> not sure if this is related to your changes.

Thanks for catching this, @Sepehr-Sobhani! I fixed the first issue, but I wasn't able to reproduce the second. Do you have a Loom?

**Comment by Sepehr-Sobhani** [comment]
> > Looks great to me, Thank you @BCerki! I just noticed two issues:
> > 
> > 1. when I remove a user from being a primary manager, I still get that project when I filter projects based on that user:
> > 
> > https://www.loom.com/share/19f8611a8a904e0cb7eee95cc28c0943
> > 
> > 2. When I add a project  I can't see the project in the project list if it doesn't have a primary manager (it's either without a manager or with a secondary manager) -> not sure if this is related to your changes.
> 
> Thanks for catching this, @Sepehr-Sobhani! I fixed the first issue, but I wasn't able to reproduce the second. Do you have a Loom?

@BCerki Please dismiss the second issue :) I was on branch`missing_projects-in-list` instead of version 2

**Review by dleard — APPROVED** [review]
Nice work! I can see projects in the list without any managers assigned. Awesome that the plpgsql function was able to be converted to pure sql.

**Pull request merged**
